### DISCUSSION TITLE
How to add buttons in the Source Control Graph view?

### DISCUSSION THREAD

**Discussion by ovidiu-vas** [microsoft/vscode-discussions#1890] [Extension Development QnA] [answered]
I'm building an extension and want to add buttons in the top menu and on each line of the "Source Control Graph". In the extension package.json file, how do I contribute to add buttons in the Source control graph? - what are the names of contribute keys?:
- title 
- each commit line
- each commit line, in the context menu (aka right click menu)

![scm-history](https://github.com/user-attachments/assets/865ebf38-1d71-4148-85af-9b8808a96cda)

Thanks!

**Comment by alefragnani** [comment]
Hi @ovidiu-vas ,

I guess there is no way to contribute on that. This view is relative new, and I suppose it was not exposed to contributions, yet.

A Quick Look in the VS Code repo (https://github.com/microsoft/vscode/issues) didn't show any issue about that, so I would suggest you to open a feature request, so the team could take a look.

Hope this helps

**Comment by gjsjohnmurray** [comment, marked as answer]
Your extension would need to use proposed APIs

src/vscode-dts/vscode.proposed.contribSourceControlHistoryTitleMenu.d.ts

src/vscode-dts/vscode.proposed.contribSourceControlHistoryitemMenu.d.ts
